webhook/webhookworker.py

```diff
@@ -0,0 +1,263 @@
+import json
+import logging
+import requests
+import time
+
+from utils.gamemechanicutil import get_raid_boss_cp
+
+log = logging.getLogger(__name__)
+
+
+class WebhookWorker:
+    # currently active ex raid mon id
+    EXRAID_MON_ID = 386
+
+    def __init__(self, args, db_wrapper):
+        self._worker_interval_sec = 10
+        self._args = args
+        self._db_wrapper = db_wrapper
+        self._last_check = int(time.time())
+
+    def __payload_type_count(self, payload):
+        count = {}
+
+        for elem in payload:
+            count[elem["type"]] = count.get(elem["type"], 0) + 1
+
+        return count
+
+    def __send_webhook(self, payload):
+        if len(payload) == 0:
+            log.info("Payload empty. Skip sending to webhook.")
+            return
+
+        # get list of urls
+        webhooks = self._args.webhook_url.replace(" ", "").split(",")
+
+        for webhook in webhooks:
+            # url cleanup
+            url = webhook.strip()
+
+            log.debug("Sending to webhook %s", url)
+            log.debug("Payload: %s" % str(payload))
+            try:
+                response = requests.post(
+                    url,
+                    data=json.dumps(payload),
+                    headers={"Content-Type": "application/json"},
+                    timeout=5,
+                )
+                if response.status_code != 200:
+                    log.warning(
+                        "Got status code other than 200 OK from webhook destination: %s"
+                        % str(response.status_code)
+                    )
+                else:
+                    log.info(
+                        "Successfully sent payload to webhook. Stats: %s",
+                        json.dumps(self.__payload_type_count(payload)),
+                    )
+            except Exception as e:
+                log.warning("Exception occured while sending webhook: %s" % str(e))
+
+    def __prepare_weather_data(self, weather_data):
+        ret = []
+
+        for weather in weather_data:
+            weather_payload = {
+                "s2_cell_id": weather["s2_cell_id"],
+                "condition": weather["gameplay_weather"],
+                "alert_severity": weather["severity"],
+                "day": weather["world_time"],
+                "time_changed": weather["last_updated"],
+                "latitude": weather["latitude"],
+                "longitude": weather["longitude"],
+            }
+
+            entire_payload = {"type": "weather", "message": weather_payload}
+
+            ret.append(entire_payload)
+
+        return ret
+
+    def __prepare_raid_data(self, raid_data):
+        ret = []
+
+        for raid in raid_data:
+            # skip ex raid mon if disabled
+            if (
+                not self._args.webhook_send_exraid
+                and raid.pokemon_id == self.EXRAID_MON_ID
+            ):
+                pass
+
+            raid_payload = {
```

Comment:
There's a few things missing here that isn't kept in RMs database but is both used in PA and other raid webhook receivers:
`park` - is a flag for if the gym is - in GMO its the `gym_details.is_ex_raid_eligible` data - this field is not saved in DB
`is_exclusive` flags if the raid is an ex raid and is "by invitation only" PA does not support this, but other tools do, and so should PA. It can be used both to filter out sending of ex raids, or to filter alerts by only ex raids (for whoever wants that). This flag is in the `gym_details.raid_info.is_exclusive` part of GMO and is not saved to the RM raid table